firestore/firestore_snippets/listen_test.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"cloud.google.com/go/firestore"
+)
+
+var duration time.Duration = 6 * time.Second
+
+func setup(ctx context.Context, t *testing.T) (*firestore.Client, string) {
+	projectID := os.Getenv("GOLANG_SAMPLES_FIRESTORE_PROJECT")
+	if projectID == "" {
+		t.Skip("Skipping firestore test. Set GOLANG_SAMPLES_FIRESTORE_PROJECT.")
+	}
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		t.Fatalf("firestore.NewClient: %v", err)
+	}
+	return client, projectID
+}
+
+func TestListen(t *testing.T) {
+	ctx := context.Background()
+	client, projectID := setup(ctx, t)
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, duration)
+	defer cancel()
+
+	// Delete all docs first to make sure setup works.
+	docs, err := client.Collection("cities").Documents(ctx).GetAll()
```

Comment:
Can we update this so we can run this test so we can run it twice at the same time in different `GOLANG_SAMPLES_PROJECT_ID`s? The usual way we've done that is by including the project ID in the collection name and making the collection name an argument for the sample functions. That way, you can get do `tc := testutil.SystemTest(t)` and use `tc.ProjectID`. That project ID is only used as an identifier, not as the `GOLANG_SAMPLES_FIRESTORE_PROJECT`.

Reply:
Thanks for clarifying it, I've made an update